### PR TITLE
Fixes error if annotation processing directories do not exist

### LINKED ISSUE #42908: Cannot initialize file manager
### Describe the bug

Live reload in dev mode fails because of generated sources
grafik

Project Setup: 
Quarkus Version: 3.13.1
Multi module maven setup with annotation processing declared in parent pom:
```
<pluginManagement>
  <plugins>
    <plugin>
      <groupId>org.apache.maven.plugins</groupId>
      <artifactId>maven-compiler-plugin</artifactId>
      <version>${compiler-plugin.version}</version>
      <configuration>
        <!-- keep parameter information for generators to work correctly -->
        <compilerArgument>-parameters</compilerArgument>
        <generatedSourcesDirectory>${project.build.directory}/generated-sources/
        </generatedSourcesDirectory>
        <source>${java.version}</source>
        <target>${java.version}</target>
        <annotationProcessorPaths>
          <path>
            <groupId>org.projectlombok</groupId>
            <artifactId>lombok</artifactId>
            <version>${lombok.version}</version>
          </path>
          <path>
            <groupId>project-groupId</groupId>
            <artifactId>annotation-processor</artifactId>
            <version>${project.version}</version>
          </path>
        </annotationProcessorPaths>
      </configuration>
    </plugin>
  </plugins>
</pluginManagement>
```

The initial startup of the app works just fine, but as soon as some file is changed and a reload is triggered i get above error message.

### Expected behavior

Live reload is working as expected with generated sources

### Actual behavior

Quarkus fails to reload. The change can be in any file for this to happen (no need to change a generated file)

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by ChristianKernDev**
Fix #42908

### CODE REVIEW

File: core/deployment/src/main/java/io/quarkus/deployment/dev/filesystem/QuarkusFileManager.java [hunk 1]
```diff
@@ -39,12 +41,22 @@ public void reset(Context context) {
...
             }
         } catch (IOException e) {
             throw new RuntimeException("Cannot reset file manager", e);
         }
     }
+    
+    private void ensureDirectories(Iterable<File> directories){
+        for (File processorPath : directories) {
+            if (!processorPath.exists()) {
+                processorPath.mkdirs();
```
Comment (FroMage) [inline core/deployment/src/main/java/io/quarkus/deployment/dev/filesystem/QuarkusFileManager.java:56]: Can you add a check for the return of `mkdirs`? It returns `false` when it fails, in which case you can throw a `RuntimeException` explaning the problem.

**Review by FroMage — CHANGES_REQUESTED** [review]

**Comment by FroMage** [comment]
Also, it looks like you need to run the formatter, you can do it with `mvn -f core/deployment process-sources`

**Comment by FroMage** [comment]
Otherwise, looks great :) Thanks

**Comment by ChristianKernDev** [comment]
Added the exception. I can't execute the formatter, because of the folling error: `org.gradle:gradle-tooling-api:jar:8.1.1 was not found`

**Review by FroMage — APPROVED** [review]
Great!

**Comment by FroMage** [comment]
> Added the exception. I can't execute the formatter, because of the folling error: org.gradle:gradle-tooling-api:jar:8.1.1 was not found

I think you have to install Gradle, you can do it with "sdk install gradle"

**Comment by FroMage** [comment]
I've tried to run the formatter locally and it didn't find anything to format, I'm going to assume your second commit fixed the issue :)

**Comment by gsmet** [comment]
Thanks!

**Pull request merged**